docs/src/components/ColorSwatchFamily/ColorSwatchFamily.component.tsx

```diff
@@ -0,0 +1,87 @@
+// REACT
+import * as React from 'react';
+// VENDOR
+import styled, { css } from 'styled-components';
+import classNames from 'classnames';
+import { ColorContrastHelpers } from '../ColorContrastHelpers';
+import { ColorSwatch } from '../ColorSwatch';
+// THEME
+import { colors, sizes } from '../../../../src/theme';
+
+
+// INTERFACES
+interface ColorSwatchFamilyProps {
+  baseColorHexCode: string;
+  baseColorName: string;
+  className?: string;
+  copyChoice: string;
+  darkColorHexCode?: string;
+  lightColorHexCode?: string;
+}
+
+
+// STYLES
+const StyledColorSwatchFamily = styled.div`
+  padding: 12px;
+  margin: 24px;
+  width: 25%;
+  min-width: 300px;
+  display: inline-block;
+  box-shadow: 0px 0px 12px 4px ${colors.ash.light};
+`;
+
+
+// JSX MARKUP
+export const ColorSwatchFamily = ({ baseColorHexCode, baseColorName, className, copyChoice, darkColorHexCode, lightColorHexCode }: ColorSwatchFamilyProps): React.ReactElement<any> => (
+  <StyledColorSwatchFamily
+    baseColorName={baseColorName}
+    baseColorHexCode={baseColorHexCode}
+    className={classNames(className)}
+    darkColorHexCode={darkColorHexCode}
+    lightColorHexCode={lightColorHexCode}
+  >
+
+    <ColorSwatch
+      className={classNames(className)}
+      backgroundColor={baseColorHexCode}
+      contrastOnBlack={ColorContrastHelpers.getRoundedColorContrast(baseColorHexCode, '#000000')}
+      contrastOnF1F1F1={ColorContrastHelpers.getRoundedColorContrast(baseColorHexCode, '#F1F1F1')}
+      contrastOnFAFAFA={ColorContrastHelpers.getRoundedColorContrast(baseColorHexCode, '#FAFAFA')}
+      contrastOnWhite={ColorContrastHelpers.getRoundedColorContrast(baseColorHexCode, '#FFFFFF')}
+      copyChoice={copyChoice}
+      name={baseColorName+'.base'}
+      textColor={ColorContrastHelpers.getHigherContrastColor(baseColorHexCode)}
+    ></ColorSwatch>
+
+    { lightColorHexCode && (
+      <ColorSwatch
```

Comment:
I think we can clean this up a little bit by separating the color swatch JSX from the logic determining what to render. [Google's implementation](https://github.com/mui-org/material-ui/blob/next/docs/src/pages/style/color/Color.js) for this is way different, but even at a high level look at how little JSX-foo is going on here.

Reply:
I'm about to push up some code with `ColorSwatch` refactored to have less props, so that should clean up this file a bit. Let me know if this still seems like an issue to you. Also, when I first started this ticket, some of the colors only had `base` defined, which is why I did that check. Perhaps we could remove if that colors file has a public contract to have `light` and `dark` also defined.